kolibri/plugins/facility/assets/src/routes.js

```diff
@@ -24,21 +24,45 @@ import {
 } from './modules/classAssignMembers/handlers';
 import { PageNames } from './constants';
 
-function facilityParamRequiredGuard(toRoute, subtopicName) {
-  if (store.getters.userIsMultiFacilityAdmin && !toRoute.params.facility_id) {
-    router.replace({
-      name: 'ALL_FACILITIES_PAGE',
-      params: { subtopicName },
-    });
-    return true;
+function facilityParamRequiredGuard(toRoute, fromRoute, subtopicName) {
+  if (
+    store.getters.userIsMultiFacilityAdmin &&
+    !toRoute.params.facility_id &&
+    fromRoute.query.subtopicName !== subtopicName
+  ) {
+    const currentSubtopicName = toRoute.query.subtopicName;
+    if (currentSubtopicName !== subtopicName) {
+      router.replace({
+        name: PageNames.ALL_FACILITIES_PAGE,
+        query: { subtopicName },
+        params: { subtopicName },
+      });
```

Comment:
I think if you just catch the error from this `router.replace` that is all that is needed along with properly defining the `subtopicName` in the ALL_FACILITIES URL.